uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/broker/BrokerCompanyPortal.java

```diff
@@ -0,0 +1,86 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.client.ui.automation.broker;
+
+import androidx.test.uiautomator.UiDevice;
+import androidx.test.uiautomator.UiObject;
+import androidx.test.uiautomator.UiObjectNotFoundException;
+import androidx.test.uiautomator.UiSelector;
+
+import com.microsoft.identity.client.ui.automation.app.App;
+
+import org.junit.Assert;
+
+import lombok.Getter;
+
+import static androidx.test.platform.app.InstrumentationRegistry.getInstrumentation;
+import static com.microsoft.identity.client.ui.automation.utils.CommonUtils.TIMEOUT;
+import static com.microsoft.identity.client.ui.automation.utils.CommonUtils.getResourceId;
+
+@Getter
+public class BrokerCompanyPortal extends App implements ITestBroker {
+
+    public final static String COMPANY_PORTAL_APP_PACKAGE_NAME = "com.microsoft.windowsintune.companyportal";
+    public final static String COMPANY_PORTAL_APP_NAME = "Intune Company Portal";
+    public final static String COMPANY_PORTAL_APK = "CompanyPortal.apk";
+
+    public BrokerCompanyPortal() {
+        super(COMPANY_PORTAL_APP_PACKAGE_NAME, COMPANY_PORTAL_APP_NAME);
+        localApkFileName = COMPANY_PORTAL_APK;
+    }
+
+    @Override
+    public void performDeviceRegistration(String username, String password) {
+        //TODO implement device registration for CP
+    }
+
+    @Override
+    public void performSharedDeviceRegistration(String username, String password) {
+        //TODO implement shared device registration for CP
+    }
+
+    @Override
+    public void handleFirstRun() {
+        //TODO handle first run for CP
```

Comment:
Do we need to file work items/throw unimplemented exceptions?

Reply:
Addressed by throwing exceptions. I will also create work items later